app/policies/post_policy.rb

```diff
@@ -0,0 +1,21 @@
+class PostPolicy < ApplicationPolicy
+  def show?
+    true
+  end
+
+  def new?
+    user.present?
```

Comment:
Before I write that you should not check the user in a policy object because it is authentication responsibility